frontend/src/hooks/use-keypair-update.tsx

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useState } from 'react'
+
+import { requestPassphrase } from '../components/passphrase-modal'
+import { AppToaster } from '../components/toaster'
+import { Intent } from '../config/intent'
+import type { GlobalActions } from '../contexts/global/global-actions'
+import type { GlobalDispatch } from '../contexts/global/global-context'
+import { useGlobal } from '../contexts/global/global-context'
+import { createLogger } from '../lib/logging'
+
+const logger = createLogger('Metadata')
+
+export type Meta = {
+  key: string
+  value: string
+}
+
+export const useKeypairUpdate = (
+  dispatch: GlobalDispatch,
+  actions: GlobalActions,
+  pubKey?: string,
+  wallet?: string
+) => {
```

Comment:
i think you're confusing implicit with explicit? implicit is when typescript infers the return type, explicit is when you declare it nevertheless. since the code uses implicit return types everywhere for all functions, i'd like to keep it that way for consistency.